### PR TITLE
feat: add pnpm support

### PR DESCRIPTION
**Pull request by shahoob**
This PR is a continuation of #161.

### CODE REVIEW

File: src/commands/new.ts [hunk 1]
```diff
@@ -130,6 +151,10 @@ export default async (name: string, flags: Record<string, boolean>) => {
 		);
 
 		await editPackageJson(response.projectName, projectName);
+
+		if (response.packageManager === 'pnpm') {
```
Comment (favna) [inline src/commands/new.ts:155]: You also need to set https://pnpm.io/npmrc#public-hoist-pattern

```
public-hoist-pattern[]=@sapphire/*
```

**Review by favna — CHANGES_REQUESTED** [review]

File: src/commands/new.ts (hunk 1 above)
Comment (favna) [reply src/commands/new.ts:155]: It's something I discovered since your previous PR btw

Comment (shahoob) [reply src/commands/new.ts:155]: Ok, will do it

Comment (shahoob) [reply src/commands/new.ts:155]: done

**Review by favna — APPROVED** [review]

**Review by enxg — APPROVED** [review]

**Pull request merged**